bin/backup.sh

```diff
@@ -35,7 +35,10 @@ if [[ -z "$NOGZIP" ]]; then
   FINAL_FILE_NAME=$BACKUP_FILE_NAME.gz
 fi
 
-/tmp/aws/bin/aws s3 cp $FINAL_FILE_NAME s3://$S3_BUCKET_PATH/$APP/$DATABASE/$FINAL_FILE_NAME
+if [[ "$S3_BUCKET_PATH" ]]; then
+  /tmp/aws/bin/aws s3 cp $FINAL_FILE_NAME s3://$S3_BUCKET_PATH/$APP/$DATABASE/$FINAL_FILE_NAME
+elif [[ "$GLACIER_VAULT" ]]; then
```

Comment:
Instead of pushing to Glacier vault directly from the script, would it make more sense to push to the s3 bucket and set a lifecycle event on the s3 bucket side?
https://aws.amazon.com/blogs/aws/archive-s3-to-glacier/
This would keep the logic in the script simple. Let me know what you think. Thanks for your help!

Reply:
With the caveat that I have no experience with Glacier, S3, or AWS...

It's a nice balance between cheap Glacier and immediately available S3, and one I didn't realize was available, thank you. However, Heroku already provides immediately available backups and rollback, so adding an S3 layer here seems redundant. I'm using this script so in the extremely unlikely event of catastrophic Heroku failure we don't lose all our data and can restore somewhere else.

I'm finding AWS overwhelming. If a few lines in a script can avoid a bunch of extra AWS steps, I'm all for it. A direct-to-Glacier option saves me work, and the script is still very, very simple.

If you decide otherwise, I can do the S3-to-Glaicer thing you suggest. If that's the case, maybe add a link to that article to the README?

Reply:
I think adding a link to readme would work. Something like this?

https://docs.aws.amazon.com/AmazonS3/latest/user-guide/create-lifecycle.html